addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ApplicationErrorScanRule.java

```diff
@@ -170,6 +170,13 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 
         } else if (!getHelper().isPage404(msg)
                 && !msg.getResponseHeader().hasContentType("application/wasm")) {
+
+            if ((!AlertThreshold.LOW.equals(this.getAlertThreshold()))
+                    && (msg.getResponseHeader().isJavaScript()
+                            || msg.getResponseHeader().isCss())) {
```

Comment:
You’d have better coverage/completeness if this leveraged commonlib’s ResourceIdentificationUtils.isJavascript or ResourceIdentificationUtils.isCss

Reply:
Done 👍

Reply:
Please add some tests that show the URL related behavior vs the existing two you added which cover the content-type behavior?

Reply:
Sorry, I couldn't understand it. Are you referring  to tests covering cases for different response codes? (like here I have made tests for 200 OK responses)

Reply:
It should not introduce duplication though.

Reply:
> It should not introduce duplication though.

Can you please elaborate? Like what kind of duplication? I am running each test first with URL defined contentType and then with contentType defined in responseBody.

Reply:
Add a new comment.